src/backend/access/transam/slru.c

```diff
@@ -764,19 +764,29 @@ SimpleLruDownloadSegment(SlruCtl ctl, int pageno, char const* path)
 
 	if (neon_use_communicator_worker) {
 		buffer = NULL;
+		pgstat_report_wait_start(WAIT_EVENT_SLRU_WRITE);
```

Comment:
I also think now for the reason those exist, and then, looking at the old code, that it utilised those strangely: while the code also performed a remote request for the SLRU segment, it didn't report waiting, but only reported so when it was copying the received bytes into a local file... I think it should have reported `wait_start` before the request just as I do here. Is it so or I am missing some logic?

Reply:
I added `pgs_r_wait_start(WE_SLRU_WRITE)` to pg_pwrite, but not to network reads, because networked reads are not writes to SLRU files.